src/commands/lab.rs

```diff
@@ -102,19 +100,101 @@ async fn handle_vscode_integration(
 ) -> Result<()> {
     is_vscode_available(pb).await?;
 
-    install_extensions(pb).await?;
-    create_vscode_settings(&global_args.project, env)?;
+    if VSCodeSettings::load()
+        .await?
+        .aqora_can_install_extensions()
+        .unwrap_or(false)
+    {
+        install_extensions(pb).await?;
+    }
 
+    create_vscode_settings(&global_args.project, env)?;
     open_vscode(global_args.project, pb).await?;
+    Ok(())
+}
 
+async fn ask_for_install_vscode_extensions(
+    allow_vscode_extensions: Option<bool>,
+    pb: &ProgressBar,
+) -> Result<()> {
+    let mut vscode_settings = VSCodeSettings::load().await?;
+
+    if let Some(allow) = allow_vscode_extensions {
+        vscode_settings
+            .set_aqora_can_install_extensions(allow)
+            .await;
+        vscode_settings.save().await?;
+        return Ok(());
+    }
+
+    fn format_extensions() -> String {
+        VSCODE_EXT
+            .iter()
+            .map(|ext| {
+                format!(
+                    "🔹 {ext}: {}",
+                    Url::parse_with_params(
+                        "https://marketplace.visualstudio.com/items",
+                        &[("itemName", *ext)]
+                    )
+                    .unwrap()
+                )
+            })
+            .collect::<Vec<_>>()
+            .join("\n")
+    }
+
+    let prompt_message = format!(
+        "\n✨ Ready to set up your VS Code environment! ✨\n\n\
+        Some essential extensions are recommended:\n\n{}\n\n\
+        Install these VS Code extensions? ",
+        format_extensions()
+    );
+
+    let can_install = tokio::task::spawn_blocking({
+        let pb = pb.clone();
+        move || {
+            pb.suspend(|| {
+                ask_confirmation(prompt_message)
+                    .map_err(|_| error::system("Failed to read input", "Please try again"))
+            })
+        }
+    })
+    .await
+    .map_err(|_| error::user("The extension installation prompt was interrupted.", ""))??;
+
+    vscode_settings
+        .set_aqora_can_install_extensions(can_install)
+        .await;
+    vscode_settings.save().await?;
     Ok(())
 }
 
+fn ask_confirmation(prompt: impl AsRef<str>) -> io::Result<bool> {
```

Comment:
please use `dialoguer` instead of rolling your own

Reply:
Ohh this things exist! Okay thank you!